src/renderers/imgui_opengl3.rs

```diff
@@ -1,3 +1,34 @@
-// TODO: Consider making local versions of gl_loader and imgui opengl renderer,
-// but this works for the time being
-pub use {gl_loader, imgui_opengl_renderer};
+use std::ffi::{c_void, CString, OsStr};
+use std::os::windows::prelude::OsStrExt;
+
+use once_cell::sync::OnceCell;
+use parking_lot::Mutex;
+use windows::core::{PCSTR, PCWSTR};
+use windows::Win32::Foundation::{FARPROC, HINSTANCE, PROC};
+use windows::Win32::Graphics::OpenGL::wglGetProcAddress;
+use windows::Win32::System::LibraryLoader::{GetProcAddress, LoadLibraryW};
+
+static mut OPENGL3_LIB: OnceCell<Mutex<HINSTANCE>> = OnceCell::new();
+
+unsafe fn get_opengl3_lib() -> HINSTANCE {
+    let mut opengl_wide_string: Vec<u16> = OsStr::new("opengl32.dll").encode_wide().collect();
+    opengl_wide_string.push(0);
+
+    LoadLibraryW(PCWSTR(opengl_wide_string.as_ptr())).unwrap()
+}
+
+pub unsafe fn get_proc_address(function_string: &str) -> *const c_void {
+    let module = OPENGL3_LIB.get_or_init(|| Mutex::new(get_opengl3_lib())).lock();
+
+    let function_c_string = CString::new(function_string).unwrap();
+    let function_name_ptr = function_c_string.as_bytes_with_nul().as_ptr();
```

Comment:
Is this cast necessary? I believe for most cases where I have used PCSTR, PCSTR(x.as_ptr() as _) or similar was sufficient, given that the function string provided (hopefully) should be null-terminated.

Reply:
Mostly just asking if you took this from some other reference or so. Honestly I don't think I've seen it before.

Reply:
Aha! No, it wasn't required at all. I'm a fresh Rust newbie (new to types as well..) so there's a lot of trial and error haha.
I believe I just looked at other C/C++ implementations + the windows-rs docs when I did this. I really had no idea what I was doing at the time - although I did get it done, albeit not so optimized 😆

Reply:
Ah I see, it is fine though 😄